### ISSUE TITLE
Postgres `read-column-thunk` will not be called if using connection pool

### ISSUE THREAD

**Issue by zhming0** [camsaul/toucan2#147]
When I use a connection pool to connect to PostgreSQL database, the `read-column-thunk` [defined here](https://github.com/camsaul/toucan2/blob/e2547454b5a5b808ca38ac9dfb264a07d2848676/src/toucan2/query_execution_backend/jdbc/postgres.clj#L18C24-L18C41) won't get triggered, causing me unable to use  the `timestamptz` type. 

The cause is that when I use a connection pool, the connection class of the query won't be `org.postgresql.jdbc.PgConnection`, instead, it will be something like `com.zaxxer.hikari.pool.HikariProxyConnection` (using Hikari as example). 

The workaround is simple yet a bit ugly: 
```
  (derive
    (Class/forName "com.zaxxer.hikari.pool.HikariProxyConnection")
    :toucan2.query-execution-backend.jdbc.postgres/connection)

```

Maybe there is a better to tackle this from toucan's end?

**Comment by camsaul** [comment]
Hi @zhming0, I implemented this in #155, please try version `1.0.531` or above and let me know if it fixes your problems.

**Comment by zhming0** [comment]
Confirmed the fix. Thank you @camsaul! 🙇🏿 🙇🏿
